R-TypeGame/Systems/Menus/Buttons/PortTextBoxInitSystem.cpp

```diff
@@ -0,0 +1,168 @@
+#include "FontPathComponent.hpp"
+#include "ButtonStateParser.hpp"
+#include "TextureRectParser.hpp"
+#include "EditableComponent.hpp"
+#include "TextLimitComponent.hpp"
+#include "Position2DComponent.hpp"
+#include "ButtonStateComponent.hpp"
+#include "TextureRectComponent.hpp"
+#include "TextPosition2DParser.hpp"
+#include "TextPosition2DComponent.hpp"
+#include "ButtonTexturePathParser.hpp"
+#include "ButtonTexturePathComponent.hpp"
+
+#define PATH_JSON "GameData/Scenes/Menus/Buttons/portTextBoxSystem.json"
+
+ButtonInitSystem::ButtonInitSystem() :
+        ASystem("ButtonInitSystem")
+{}
+
+static void handleThis(ECS::Registry& reg, int idxPacketEntities)
+{
+    std::shared_ptr<ButtonStateComponent> state = std::dynamic_pointer_cast<ButtonStateComponent>(reg.get_components<IComponent>("ButtonStateComponent")[idxPacketEntities]);
+    std::shared_ptr<TextComponent> text = std::dynamic_pointer_cast<TextComponent>(reg.get_components<IComponent>("TextComponent")[idxPacketEntities]);
+    std::shared_ptr<DefaultTextComponent> defaultText = std::dynamic_pointer_cast<DefaultTextComponent>(reg.get_components<IComponent>("DefaultTextComponent")[idxPacketEntities]);
+    if (state && text && defaultText) {
+        if (state->state == ButtonStateComponent::ButtonState::CLICKED) {
+            state->state = ButtonStateComponent::ButtonState::NONE;
+            if (text->text.empty())
+                text->text = defaultText->text;
+        } else
+            state->state = ButtonStateComponent::ButtonState::CLICKED;
+    }
+}
+
+static void handleOther(ECS::Registry& reg, int idxPacketEntities)
+{
+    ECS::SparseArray<IComponent> states = reg.get_components<IComponent>("ButtonStateComponent");
+    ECS::SparseArray<IComponent> callbacks = reg.get_components<IComponent>("CallBackComponent");
+    ECS::SparseArray<IComponent> texts = reg.get_components<IComponent>("TextComponent");
+    ECS::SparseArray<IComponent> defaultTexts = reg.get_components<IComponent>("DefaultTextComponent");
+    for (int i = 0; i < states.size() && i < callbacks.size() && i < texts.size() && i < defaultTexts.size(); i++) {
+        std::shared_ptr<ButtonStateComponent> newState = std::dynamic_pointer_cast<ButtonStateComponent>(states[i]);
+        std::shared_ptr<CallBackComponent> callback = std::dynamic_pointer_cast<CallBackComponent>(callbacks[i]);
+        std::shared_ptr<TextComponent> newText = std::dynamic_pointer_cast<TextComponent>(texts[i]);
+        std::shared_ptr<DefaultTextComponent> newDefaultText = std::dynamic_pointer_cast<DefaultTextComponent>(defaultTexts[i]);
+        if (!newState || !callback || !newText || !newDefaultText)
+            continue;
+        if (newState->state == ButtonStateComponent::ButtonState::CLICKED && i != idxPacketEntities) {
+            newState->state = ButtonStateComponent::ButtonState::NONE;
+            if (newText->text.empty())
+                newText->text = newDefaultText->text;
+        }
```

Comment:
Line 49 you check if all are not null to verify if all are initialised, you can keep the same logic

Reply:
I prefer (and @mathieurobert1 too) the logic of the line 70, so I changed the line 49 to keep the same logic on both.
The main argument is that the line 70 is more readable and avoid condition nesting.